folding-schemes-solidity/templates/kzg10_verifier.askama.sol

```diff
@@ -0,0 +1,263 @@
+{{ sdpx }}
+
+{{ pragma_version }}
+
```

Comment:
Should this contain some reference to the original code and its license?

Reply:
Indeed, added a few comments on the code inspiration and the changes we did to it. See [8c843e5](https://github.com/privacy-scaling-explorations/folding-schemes/pull/70/commits/8c843e5a46ce4f7f739473f898b9d248390e3dc5)

The sdpx statement is templated because`solc` will not compile a solidity contract which contains conflicting licenses. In our case, we have the `kzg10` and `groth16` verifiers co-habitating for the final decider, one is licensed MIT and the other gpl-3. 

Templating allows us to just provide `None` for now and default to gpl-3 for the final decider contract, reminding ourselves that we should make a decision on that front!